### PR TITLE
feat: [PL-18548]: ng-auth-ui support in local environment

### PR DESCRIPTION
**Pull request by avinashmadhwani02**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -30,6 +30,9 @@ const moduleFederationConfig = require('./configs/modulefederation.config.js')
 const ExternalRemotesPlugin = require('external-remotes-plugin')
 
 const DEV = process.env.NODE_ENV === 'development'
+const HARNESS_ENABLE_NG_AUTH_UI = process.env.HARNESS_ENABLE_NG_AUTH_UI
```
Comment (abhinavrastogi-harness) [inline webpack.config.js:33]: if the value of `process.env.HARNESS_ENABLE_NG_AUTH_UI` is `false`, we will still enable it with this logic.
you should explicitely check `typeof` probably.

File: webpack.devServerProxy.config.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ require('dotenv').config()
 
 const baseUrl = process.env.BASE_URL ?? 'https://qa.harness.io/gateway'
 const targetLocalHost = (process.env.TARGET_LOCALHOST && JSON.parse(process.env.TARGET_LOCALHOST)) ?? true // set to false to target baseUrl environment instead of localhost
-console.table({ baseUrl, targetLocalHost })
+console.table({ baseUrl, targetLocalHost, HARNESS_ENABLE_NG_AUTH_UI: process.env.HARNESS_ENABLE_NG_AUTH_UI || true })
```
Comment (abhinavrastogi-harness) [inline webpack.devServerProxy.config.js:12]: move this new log item to webpack.config.js.

File: src/index.html [hunk 3]
```diff
@@ -40,11 +35,21 @@
...
     <script>
       window.HARNESS_ENABLE_NG_AUTH_UI = 'HARNESS_ENABLE_NG_AUTH_UI_PLACEHOLDER' === 'true'
       window.HARNESS_ENABLE_FULL_STORY = 'HARNESS_ENABLE_FULL_STORY_PLACEHOLDER' === 'true'
       window.HARNESS_ENABLE_APPDY_EUM = 'HARNESS_ENABLE_APPDY_EUM_PLACEHOLDER' === 'true'
     </script>
+    <% if (__DEV__) { %>
+    <script>
+      window.deploymentType = 'SAAS'
+    </script>
+    <% } %> <% if (HARNESS_ENABLE_NG_AUTH_UI) { %>
```
Comment (abhinavrastogi-harness) [inline src/index.html:48]: new `if` should be on a new line

File: src/modules/10-common/RouteDestinations.tsx [hunk 4]
```diff
@@ -43,6 +44,11 @@ const justAccountPath = withAccountId(() => '/')
 
 export default (
   <>
+    {__DEV__ && (
+      <Route path="/account/:accountId/dashboard">
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/RouteDestinations.tsx:48]: add a comment in the code explaining why this is needed

File: webpack.config.js (hunk 1 above)
Comment (avinashmadhwani02) [reply webpack.config.js:33]: It will not be enabled because `false` will be a string. Changing it to avoid confusion

**Review by abhinavrastogi-harness — DISMISSED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/modules/10-common/RouteDestinations.tsx [hunk 5]
```diff
@@ -43,6 +44,13 @@ const justAccountPath = withAccountId(() => '/')
 
 export default (
   <>
+    {__DEV__ && (
+      // Redirecting users whose default experience is CG from CG dashboard route to NG home page.
```
Comment (reetika04) [inline src/modules/10-common/RouteDestinations.tsx:48]: whose default experience is NG?

Comment (avinashmadhwani02) [reply src/modules/10-common/RouteDestinations.tsx:48]: Should be CG right?

**Review by reetika04 — APPROVED** [review]

**Pull request merged**